### PR TITLE
Added a button to 'select all' of the component checkboxes on the download page

### PR DESCRIPTION
**Pull request by KiriHoy**
Added a button to 'select all' of the component checkboxes on the download page instead of having to individually select each one. 
#458

### CODE REVIEW

File: src/js/footer/050-furnace-code.js [hunk 1]
```diff
@@ -2,8 +2,7 @@ var furnaceComponents = document.querySelectorAll( '.js-furnace-selector' );
 var furnaceNPM        = document.querySelectorAll( '.js-furnace-code code' )[ 0 ];
 var furnaceClear      = document.querySelectorAll( '.js-furnace-clear ' )[ 0 ];
 var furnaceDownload   = document.querySelectorAll( '.furnace__buildbox__download' )[ 0 ];
-
-
+var  selectAll    	  = document.querySelector('.furnace__selectall');
```
Comment (adamzerella) [inline src/js/footer/050-furnace-code.js]:
```suggestion
var  selectAll   = document.querySelector('.furnace__selectall');	 
```

File: src/layout/download/furnace.js [hunk 2]
```diff
@@ -51,6 +51,7 @@ const Furnace = ({ components, _ID, _body, _parseYaml, _relativeURL }) => {
 					<h3 className="furnace__title">
 						Select components
 						<button type="reset" className="js-furnace-clear furnace__clear au-btn au-btn--tertiary icon icon--right icon--close">Clear selections</button>
```
Comment (adamzerella) [inline src/layout/download/furnace.js]:
```suggestion
						<button type="reset" className="js-furnace-clear furnace__clear au-btn au-btn--tertiary icon icon--right icon--close">Reset</button>
```

@sukhrajghuman Not sure on the wording for this one, Chameleon uses text: `Reset`

File: src/js/footer/050-furnace-code.js [hunk 3]
```diff
@@ -64,3 +63,13 @@ function ToggleNPM( $this ) {
...
 	furnaceNPM.innerHTML = 'npm install --save @gov.au/core';
 }
+
+/**
+ * Select all button selects all checkbox elements on page
+ */ 
+AddEvent( selectAll, 'click', function( event ){
+	furnaceComponents.forEach(function(furnaceComponents) {
+		furnaceComponents.checked=true;
+});
```
Comment (adamzerella) [inline src/js/footer/050-furnace-code.js]:
```suggestion
	});
```

File: src/js/footer/050-furnace-code.js [hunk 4]
```diff
@@ -64,3 +63,13 @@ function ToggleNPM( $this ) {
...
+
```
Comment (adamzerella) [inline src/js/footer/050-furnace-code.js]:
```suggestion
```

File: src/js/footer/050-furnace-code.js [hunk 5]
```diff
@@ -64,3 +63,13 @@ function ToggleNPM( $this ) {
 function ClearNPM() {
 	furnaceNPM.innerHTML = 'npm install --save @gov.au/core';
 }
+
+/**
+ * Select all button selects all checkbox elements on page
+ */ 
+AddEvent( selectAll, 'click', function( event ){
+	furnaceComponents.forEach(function(furnaceComponents) {
```
Comment (adamzerella) [inline src/js/footer/050-furnace-code.js]: I would probably rename the callback argument `furnaceComponents` to something a bit less confusing like `element` or `item`. Having the array the same name as the iterated item reads a bit funny

File: src/js/footer/050-furnace-code.js [hunk 6]
```diff
@@ -2,8 +2,7 @@ var furnaceComponents = document.querySelectorAll( '.js-furnace-selector' );
 var furnaceNPM        = document.querySelectorAll( '.js-furnace-code code' )[ 0 ];
 var furnaceClear      = document.querySelectorAll( '.js-furnace-clear ' )[ 0 ];
 var furnaceDownload   = document.querySelectorAll( '.furnace__buildbox__download' )[ 0 ];
-
-
+var  selectAll   = document.querySelector('.furnace__selectall');
```
Comment (adamzerella) [inline src/js/footer/050-furnace-code.js]: I might rename this as well to something like `furnaceSelectBtn`

File: src/js/footer/050-furnace-code.js [hunk 7]
```diff
@@ -64,3 +63,13 @@ function ToggleNPM( $this ) {
 function ClearNPM() {
 	furnaceNPM.innerHTML = 'npm install --save @gov.au/core';
 }
+
```
Comment (adamzerella) [inline src/js/footer/050-furnace-code.js]:
```suggestion


```

File: src/sass/components/_furnace.scss [hunk 8]
```diff
@@ -1,13 +1,28 @@
-.au-body .furnace__title {
+.furnace__panel {
+	 .furnace__title {
```
Comment (sukhrajghuman) [inline src/sass/components/_furnace.scss]:
```suggestion
	 .furnace__title {
	 
```

File: src/sass/components/_furnace.scss [hunk 9]
```diff
@@ -1,13 +1,28 @@
...
+		//display: inline;
+		}
+	}
 	.furnace__clear {
-		@include AU-space( padding, 0.5unit 0 );
-		@include AU-space( margin-top, -0.5unit );
-		float: right;
+		//@include AU-space( padding, 0.5unit 0 );
+		//@include AU-space( margin-top, -0.5unit );
+		//float: right;
```
Comment (sukhrajghuman) [inline src/sass/components/_furnace.scss]: Do we need these comments?

File: src/js/footer/050-furnace-code.js [hunk 10]
```diff
@@ -64,3 +64,13 @@ function ToggleNPM( $this ) {
...
+
+/**
+ * Selects all component checkboxes on page
+ */ 
+AddEvent( furnaceSelectAll, "click", function( event, $this ) {
```
Comment (sukhrajghuman) [inline src/js/footer/050-furnace-code.js]:
```suggestion
AddEvent( furnaceSelectAll, "click", function() {
```

**Pull request merged**